src/common.c

```diff
@@ -1418,16 +1418,81 @@ sr_create_module_imps_incs_r(const struct lys_module *ly_mod)
     return NULL;
 }
 
+static sr_error_info_t *
+sr_shm_prefix(const char **prefix)
+{
+    sr_error_info_t *err_info = NULL;
+
+    *prefix = getenv(SR_SHM_PREFIX_ENV);
+    if (*prefix == NULL) {
+        *prefix = SR_SHM_PREFIX_DEFAULT;
+    } else if (strchr(*prefix, '/') != NULL) {
+        *prefix = NULL;
+        sr_errinfo_new(&err_info, SR_ERR_INVAL_ARG, NULL,
+            SR_SHM_PREFIX_ENV" cannot contain slashes.");
+    }
+
+    return err_info;
+}
+
+sr_error_info_t *
+sr_path_main_shm(char **path)
+{
+    sr_error_info_t *err_info = NULL;
+    const char *prefix;
+
+    err_info = sr_shm_prefix(&prefix);
+    if (err_info) {
+        goto end;
+    }
+
+    if (asprintf(path, "/%s_main", prefix) == -1) {
+        SR_ERRINFO_MEM(&err_info);
+        *path = NULL;
+    }
+
+end:
```

Comment:
Please do not use redundant labels, this one is not needed at all.

Reply:
I removed it.